ResearchSuite/ResearchSuite/RSDFactory.swift

```diff
@@ -70,6 +70,25 @@ open class RSDFactory {
     
     
     // MARK: Task factory
+    
+    /**
+     Use the resource transformer to get a data object to decode into a task.
+     
+     @param resourceTransformer     The resource transformer.
+     
+     @return                        The decoded task.
+     */
+    open func decodeTask(with taskInfo: RSDTaskInfoStepObject) throws -> RSDTask {
+        guard let resourceTransformer = taskInfo.taskTransformer as? RSDTaskResourceTransformer else {
+            throw RSDValidationError.invalidType("Cannot get embedded resource.")
+        }
+        let (data, type) = try resourceTransformer.resourceData()
```

Comment:
should maybe just call the func below with the resourceTransformer, taskInfo, and nil (or leave out) schemaInfo?